validation.go

```diff
@@ -1,5 +1,28 @@
 package fyne
 
+// ExpandedValidator expands on the Validatable interface to allow the widget to expose more context to the parent.
+//
+// Since: 2.5
+type ExpandedValidator interface {
+	Validatable
+
+	// HasValidator makes it possible for a parent that cares about child validation (e.g. widget.Form)
+	// to know if the developer has attached a validator or not.
+	HasValidator() bool
+
+	// SetOnFocusChanged is intended for parent widgets or containers to hook into the change of focus.
+	// The function might be overwritten by a parent that cares about child validation (e.g. widget.Form).
+	SetOnFocusChanged(func(bool))
```

Comment:
This feels like exposing internal details - I can see why it was needed but I don't think it's the cleanest way.
Can't we wrap the widgets if they are already `Focusable` and intercept the `FocusGained` and `FocusLost`?